controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,728 @@
+	if err != nil {
+		return nil, fmt.Errorf("couldn't authenticate to Azure cloud")
+	}
+
+	return autorest.NewBearerAuthorizer(spToken), nil
+}
+
+func newResourceGroupsClient(authorizer autorest.Authorizer, spec aksv1.AKSClusterConfigSpec) (*resources.GroupsClient, error) {
+	if authorizer == nil {
+		return nil, fmt.Errorf("error getting Azure authorizer")
+	}
+
+	baseURL := spec.BaseURL
+	if baseURL == "" {
+		baseURL = azure.PublicCloud.ResourceManagerEndpoint
+	}
+
+	client := resources.NewGroupsClientWithBaseURI(baseURL, spec.SubscriptionID)
+	client.Authorizer = authorizer
+
+	return &client, nil
+}
+
+func createResourceGroup(ctx context.Context, groupsClient *resources.GroupsClient, spec aksv1.AKSClusterConfigSpec) (bool, error) {
+	future, err := groupsClient.CreateOrUpdate(
+		ctx,
+		spec.ResourceGroup,
+		resources.Group{
+			Name:	  to.StringPtr(spec.ResourceGroup),
+			Location: to.StringPtr(spec.ResourceLocation),
+		})
+
+	if err != nil {
+		return false, fmt.Errorf("[AKS] error creating resource group '%s': %v", spec.ResourceGroup, err)
+	}
+
+	return future.StatusCode == 201, nil
+}
+
+func checkResourceGroup(ctx context.Context, groupsClient *resources.GroupsClient, spec aksv1.AKSClusterConfigSpec) (bool, error) {
+	resp, err := groupsClient.CheckExistence(ctx, spec.ResourceGroup)
+	if err != nil {
+		return false, fmt.Errorf("[AKS] error getting resource group '%s': %v", spec.ResourceGroup, err)
+	}
+
+	return resp.StatusCode == 204, nil
+}
+
+func removeResourceGroup(ctx context.Context, groupsClient *resources.GroupsClient, spec aksv1.AKSClusterConfigSpec) error {
+	future, err := groupsClient.Delete(ctx, spec.ResourceGroup)
+	if err != nil {
+		return fmt.Errorf("[AKS] error removing resource group '%s': %v", spec.ResourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, groupsClient.Client); err != nil {
+		logrus.Errorf("[AKS] cannot get the AKS cluster create or update future response: %v", err)
+		return err
+	}
+
+	logrus.Infof("Cluster %v removed successfully", spec.ClusterName)
+
+	return nil
+}
+
+func newClustersClient(authorizer autorest.Authorizer, spec aksv1.AKSClusterConfigSpec) (*containerservice.ManagedClustersClient, error) {
+	if authorizer == nil {
+		return nil, fmt.Errorf("error getting Azure authorizer")
+	}
+
+	baseURL := spec.BaseURL
+	if baseURL == "" {
+		baseURL = azure.PublicCloud.ResourceManagerEndpoint
+	}
+
+	client := containerservice.NewManagedClustersClientWithBaseURI(baseURL, spec.SubscriptionID)
+	client.Authorizer = authorizer
+
+	return &client, nil
+}
+
+// CreateAKS creates a new managed Kubernetes cluster
+func createOrUpdateCluster(ctx context.Context, secretsCache wranglerv1.SecretCache, clusterClient *containerservice.ManagedClustersClient, spec aksv1.AKSClusterConfigSpec) (c containerservice.ManagedCluster, err error) {
+
+	dnsPrefix := spec.DNSPrefix
+	if dnsPrefix == "" {
+		dnsPrefix = spec.ClusterName + "-master"
+	}
```

Comment:
A nit here - why add `-master`? `dnsPrefix` is just a prefix for the node dns host name.

Reply:
It was leftover from old driver, it was removed